sw/device/sca/ecc256_keygen_serial.c

```diff
@@ -144,6 +198,71 @@ static void p256_run_keygen(uint32_t mode, const uint32_t *seed,
   sca_set_trigger_low();
 }
 
+static void ecc256_ecdsa_secret_keygen_batch(const uint8_t *data,
+                                             size_t data_len) {
+  uint32_t num_traces = 0;
+  uint32_t out[kEcc256SeedNumWords];
+  uint32_t batch_digest[kEcc256SeedNumWords];
+  uint8_t dummy[kEcc256SeedNumBytes];
+  SS_CHECK(data_len == sizeof(num_traces));
+  num_traces = read_32(data);
+
+  if (num_traces > kNumBatchOpsMax) {
+    LOG_ERROR("Too many traces for one batch.");
+    return;
+  }
+
+  // zero the batch digest
+  for (uint32_t j = 0; j < kEcc256SeedNumWords; ++j) {
+    batch_digest[j] = 0;
+  }
+
+  for (uint32_t i = 0; i < num_traces; ++i) {
+    if (run_fixed) {
+      memcpy((uint8_t *)batch_seeds[i], (uint8_t *)ecc256_seed,
+             kEcc256SeedNumBytes);
+    } else {
+      // One additional PRNG run required to be in sync with host
```

Comment:
On the host side the `kpt.next()` iterator is used. This iterator outputs two numbers (thus requires two runs of the PRNG). I **think** this can be optimized on the host side by using `kpt.next_key()` instead, which **should** only require on run of the PRNG.

Reply:
FYI: A more detailed explanation of this is given by Michael in https://github.com/lowRISC/ot-sca/issues/116

Reply:
Looks like this is an easy fix, that we could include in this PR. 
@wettermo I'll send you a patch

Reply:
Done, included the fix! Also tested it, runs fast and smoothly :)